code/game/objects/items/devices/flashlight.dm

```diff
@@ -198,6 +198,75 @@
 	if(creator)
 		visible_message("<span class='danger'>[creator] created a medical hologram!</span>")
 
+/obj/item/flashlight/crank
+	name = "crank flashlight"
+	desc = "An older flashlight model, used in emergency situations."
+	icon_state = "crank"
+	light_range = 3  // old
+	light_power = 2
+	actions_types = list(/datum/action/item_action/toggle_light, /datum/action/item_action/crank)
+
+	var/current_charge = 0
+	var/max_charge = 300  // 5 minutes
+	var/crank_adds = 30 // 10 cranks to full
+
+/obj/item/flashlight/crank/Initialize(mapload)
+	current_charge = max_charge
+	return ..()
+
+/obj/item/flashlight/crank/update_brightness(mob/user)
+	if(current_charge <= 0)
+		user.visible_message("<span class='warning'>The flashlight flickers a few times, before cutting out.</span>")
+		on = FALSE
+		current_charge = 0  // underflow prevent
```

Comment:
return here? Why do you want to process 0 charge?

Reply:
so it updates the brightness through the . = ..()